Add structured logging on FXGraphCache hit

jamesjwu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129588


We'll also want to do this for AOTAutogradCache once that's ready 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D59144226](https://our.internmc.facebook.com/intern/diff/D59144226)

jamesjwu committed:
Update

[ghstack-poisoned]

oulgen commented on torch/_inductor/codecache.py:
```diff
@@ -902,6 +902,10 @@ def iterate_over_candidates() -> Generator[CompiledFxGraph, None, None]:
             # Not expected, but in case the PyCodeCache entry is removed from
             # underneath us, treat it as a cache miss and recompile.
             log.error("Failed to load cached artifact: %s", artifact_path)
+            trace_structured(
```
we dont need this, in this case we will compile as if no cache hit

oulgen commented on torch/_inductor/codecache.py:
```diff
@@ -923,6 +927,12 @@ def iterate_over_candidates() -> Generator[CompiledFxGraph, None, None]:
+        # On cache hit, use artifact path as filename
+        trace_structured(
+            "inductor_output_code",
+            lambda: {"filename": artifact_path},
```
should we add something to indicate that this was a result of cache hit?

jamesjwu replied:
I don't want to change the inductor_output_code interface just for FXGraphCache: I think it belongs in CompilationMetrics instead. We can work on some cache stats in compilation metrics to include both AOTAutogradCache and FXGraphCache, and log that directly via tlparse's compilation metrics parsing.

jamesjwu committed:
Update

[ghstack-poisoned]

jamesjwu commented:
@pytorchbot rebase

oulgen commented:
@pytorchbot merge

oulgen committed:
Update on "Add structured logging on FXGraphCache hit"



We'll also want to do this for AOTAutogradCache once that's ready 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

oulgen commented:
@pytorchbot merge

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59144226).

oulgen commented:
@pytorchbot merge -f "flaky tests"
